specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2022-10-01-preview/grafana.json

```diff
@@ -0,0 +1,1998 @@
+      "type": "object",
+      "properties": {
+        "saasSubscriptionDetails": {
+          "$ref": "#/definitions/SaasSubscriptionDetails"
+        },
+        "marketplaceTrialQuota": {
+          "$ref": "#/definitions/MarketplaceTrialQuota"
+        }
+      }
+    },
+    "GrafanaPlugin": {
+      "description": "Plugin of Grafana",
+      "type": "object",
+      "properties": {
+        "pluginId": {
+          "description": "Grafana plugin id",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "GrafanaAvailablePlugin": {
+      "description": "Available plugins of grafana",
+      "type": "object",
+      "properties": {
+        "pluginId": {
+          "description": "Grafana plugin id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Grafana plugin display name",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "GrafanaAvailablePluginListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GrafanaAvailablePlugin"
+          },
+          "x-ms-identifiers": [
+            "pluginId"
+          ]
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "SaasSubscriptionDetails": {
+      "description": "SaaS subscription details of a Grafana instance",
+      "type": "object",
+      "properties": {
+        "planId": {
+          "type": "string"
+        },
+        "offerId": {
+          "type": "string"
+        },
+        "publisherId": {
+          "type": "string"
+        },
+        "term": {
+          "$ref": "#/definitions/SubscriptionTerm"
+        }
+      }
+    },
+    "SubscriptionTerm": {
+      "description": "The current billing term of the SaaS Subscription.",
+      "type": "object",
+      "properties": {
+        "termUnit": {
+          "type": "string"
+        },
+        "startDate": {
+          "description": "The date and time in UTC of when the billing term starts.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "endDate": {
+          "description": "The date and time in UTC of when the billing term ends.",
+          "type": "string",
+          "format": "date-time"
+        }
+      }
+    },
+    "MarketplaceTrialQuota": {
+      "description": "The allocation details of the per subscription free trial slot of the subscription.",
+      "type": "object",
+      "properties": {
+        "availablePromotion": {
+          "description": "Available enterprise promotion for the subscription",
+          "$ref": "#/definitions/AvailablePromotion"
+        },
+        "grafanaResourceId": {
+          "description": "Resource Id of the Grafana resource which is doing the trial.",
+          "type": "string"
```

Comment:
Use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details) here.

Reply:
[must fix]

Reply:
Will fix in current version